lib/did-io.js

```diff
@@ -194,22 +194,34 @@ export function _methodById({doc, methodId}) {
 }
 
 /**
- * Parses the DID into various component (currently, only cares about prefix).
+ * Parses the DID into scheme and method component.
  *
  * @example
  * parseDid({did: 'did:v1:test:nym'});
- * // -> {prefix: 'v1'}
+ * // -> {method: 'v1'}
  *
  * @param {string} did - DID uri.
  *
- * @returns {{prefix: string}} Returns the method prefix (without `did:`).
+ * @returns {{scheme: string, method: string}} Returns the scheme & method.
  */
 export function parseDid({did}) {
   if(!did) {
     throw new TypeError('DID cannot be empty.');
   }
 
-  const prefix = did.split(':').slice(1, 2).join(':');
+  const [scheme, method] = did.split(':');
 
-  return {prefix};
+  return {scheme, method};
+}
+
+/**
+ * Takes in a did and determines if it has url characters in it.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.did - A DID.
+ *
+ * @returns {boolean} Does the did potentially have a query or a fragment in it?
+ */
+export function isDidUrl({did}) {
```

Comment:
This naming is a bit confusing at first glance.  I assume this is trying to check between [DID Syntax](https://www.w3.org/TR/did-core/#did-syntax) and [DID URL Syntax](https://www.w3.org/TR/did-core/#did-url-syntax)?  It looks like the `path-abempty`, query, and fragment parts are all optional so shouldn't every regular DID be a DID URL too, even if the regex below doesn't pass?

Reply:
All DIDs are valid DID URLs, but not all DID URLS are valid DIDs. A DID URL can for instance end with `:`.